kpi/models/object_permission.py

```diff
@@ -1044,17 +1044,17 @@ def build_dict(user_id_, object_permissions_):
         if user is not None:
             user_id = user.pk if not user.is_anonymous \
                 else settings.ANONYMOUS_USER_ID
```

Comment:
IMHO, assigning `user_id` is useless now and more confusing?

What about:
```python3
if user is not None:
    # Retrieve anonymous' permissions first
    all_anon_object_permissions = self.__get_all_user_permissions(
          user_id=settings.ANONYMOUS_USER_ID
    )
    perms = build_dict(settings.ANONYMOUS_USER_ID, 
                       all_anon_object_permissions.get(self.pk))

    if not user.is_anonymous:
        all_object_permissions = self.__get_all_user_permissions(
             user_id=user.pk
        )
        perms += build_dict(user.pk, all_object_permissions.get(self.pk))
```

Reply:
I agree 👍